bin/bulk-update-apps.mjs

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env node
+
+/**
+ * @file
+ * You can use this script to bump the version of many apps with the same changes.
+ * You need to specify a hash for a git commit along with a message that describes the changes made to the apps.
+ *
+ * The apps that were changed in the git commit are bumped to a new minor version,
+ * and their changelogs are updated with the message you provided.
+ *
+ * Run it like this:
+ * node bin/bulk-update-apps.mjs GITHASH CHANGELOGMESSAGE
+ *
+ * Example command:
+ * node bin/exempt-lint.mjs 29ced17i7 'Minor code improvements'
+ *
+ * You can also run it in output mode like this:
+ * node bin/bulk-update-apps.mjs GITHASH --output
+ *
+ * This mode doesn't make any changes to your files, it outputs the ID's of all apps that would be bumped.
+ */
+
+import { exec } from "node:child_process";
+import fs from "node:fs/promises";
+
+const commitHash = process.argv[2];
+if (!commitHash || commitHash === "--output") {
+  throw new Error(
+    "First argument needs to be a git commit hash, something like '29ced17i7'",
+  );
+}
+
+const changelogMessage = process.argv[3];
+if (!changelogMessage) {
+  throw new Error(
+    "Second argument needs to be a changelog message, something like 'Minor code improvements'",
+  );
+}
+
+let outputFlag = false;
+if (process.argv[3] === "--output" || process.argv[4] === "--output") {
+  outputFlag = true;
+}
+
+const gitOutput = await new Promise((resolve) => {
+  exec(
+    `git diff-tree --no-commit-id --name-only ${commitHash} -r`,
```

Comment:
What if the user passes `commitHash` as `; rm -rf ~/` ?

Is there an execl / execv function we can use? (one that doesn't invoke `sh`)

Reply:
Again, since it is only used by us locally, we will probably be fine. Just don't do command injection on your own machine.

I don't think there is a safe way to run commands in Nodejs apart from using a library, but a quick solution to this could be to check that the input only contains a-z and 0-9 characters.